integration-tests/utils/integration-utils.js

```diff
@@ -191,10 +191,30 @@ const dismissNotifications = async page => {
   return !!notificationCloseButtons.length && delay(1000) // delayed for alerts to animate off
 }
 
+const loadSignInPageAndWaitForReady = (page, url, timeout = 30 * 1000) => {
+  return Promise.all([
+    page.goto(url, waitUntilLoadedOrTimeout(timeout)),
+    Promise.race([
+      page.waitForXPath('//*[@id="signInButton"]', { visible: true, timeout }),
+      page.waitForXPath('//a[@href="#workspaces"]', { visible: true, timeout })
+    ])
+  ])
+    .then(() => waitForNoSpinners(page))
+}
+
 const signIntoTerra = async (page, { token, testUrl }) => {
-  !!testUrl && await page.goto(testUrl, waitUntilLoadedOrTimeout(60 * 1000))
-  await page.waitForXPath('//*[contains(normalize-space(.),"Loading Terra")]', { hidden: true, timeout: 60 * 1000 })
-  await waitForNoSpinners(page)
+  if (!!testUrl) {
+    try {
+      console.log(`Loading page: ${testUrl}`)
+      await loadSignInPageAndWaitForReady(page, testUrl)
+    } catch (err) {
+      console.error(err)
+      console.error(`Error: Page loading timed out during sign in. Reloading URL: "${testUrl}"`)
+      await page.reload({ waitUntil: 'load' })
+      await loadSignInPageAndWaitForReady(page, testUrl)
```

Comment:
Does `page.goto` in this function cause the page to be loaded again after `page.reload`?

Reply:
😞 yeah, looks that way. How about the updated code?